R/compare.R

```diff
@@ -314,6 +329,55 @@ expand_diff_text <- function(old, new) {
   list(old = old, new = new)
 }
 
+merge_packages <- function(old, new) {
+  opkgs <- parse_pkgs(old)
+  npkgs <- parse_pkgs(new)
+
+  if (is.null(opkgs) || is.null(opkgs)) return(list(old = old, new = new))
+
+  names_to_keep <- c("package", "version", "source")
+  opkgs$pkgs <- opkgs$pkgs[names_to_keep]
+  npkgs$pkgs <- npkgs$pkgs[names_to_keep]
+  names(opkgs$pkgs) <- c("package", "old_version", "old_source")
+  names(npkgs$pkgs) <- c("package", "new_version", "new_source")
+
+  merge_res <- merge(opkgs$pkgs, npkgs$pkgs, all = TRUE)
+  merge_res <- with(merge_res,
+       data.frame(
+         package = package,
+         "v!=" = mark(is_different(old_version, new_version)),
+         old_version = old_version,
+         new_version = new_version,
+         "s!=" = mark(is_different(old_source, new_source)),
+         old_source = old_source,
+         new_source = new_source,
+         stringsAsFactors = FALSE,
+         row.names = NULL,
+         check.names = FALSE
+       )
+  )
+
+  oldopts <- options(cli.num_colors = 1)
+  on.exit(options(oldopts), add = TRUE)
+  fmt <- format_df(merge_res)
+
+  old <- insert_instead(old, opkgs$begin, opkgs$end, fmt)
+  new <- insert_instead(new, npkgs$begin, npkgs$end, fmt)
```

Comment:
This is what I meant by "hack-y, but the best way to get the desired behaviour without a larger refactoring".

I get the merge output, ultimately, by creating it here, inserting same lines into both `old` and `new` and letting `cli::diff_chr()` get called on it, which becomes a no-op.

Reply:
Yeah, that is hack-y indeed. Let's not worry about it now, though.